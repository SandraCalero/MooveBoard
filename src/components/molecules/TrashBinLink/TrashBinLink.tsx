import { Link } from 'react-router-dom';
import trashBinEmpty from 'assets/icons/trashBinEmpty.png';
/* import trashBinFull from 'assets/icons/trashBinFull.png'; */

export default function TrashBinLink() {
	return (
		<Link
			to='/trashbin'
			className='link'
			onDrop={(event) => console.log(event)}
			onDragOver={(event) => console.log(event)}
		>
			<img src={trashBinEmpty} alt='Trash bin' />
			<span>Trash Bin</span>
			{/* <img src={trashBinFull} alt='Trash bin' /> */}
		</Link>
	);
}
