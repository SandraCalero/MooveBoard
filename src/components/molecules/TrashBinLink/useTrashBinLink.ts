import { useSelector } from 'react-redux';
import { RootState } from 'redux/store';
import trashBinEmpty from 'assets/icons/trashBinEmpty.png';
import trashBinFull from 'assets/icons/trashBinFull.png';

const useTrashBinLink = () => {
	const deletedPostIts = useSelector(
		(state: RootState) => state.trashspace.deletedPostIts
	);
	const isEmpty = !deletedPostIts.length;
	const trashIcon = isEmpty ? trashBinEmpty : trashBinFull;

	return {
		trashIcon,
		isEmpty,
	};
};

export default useTrashBinLink;
