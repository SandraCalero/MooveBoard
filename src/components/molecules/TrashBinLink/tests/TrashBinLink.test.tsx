import { BrowserRouter } from 'react-router-dom';
import { render } from 'utils/tests/test-utils';
import TrashBinLink from '../TrashBinLink';

describe('<TrashBinLink />', () => {
	it('Snapshot', () => {
		const { container, getByText } = render(
			<BrowserRouter>
				<TrashBinLink />
			</BrowserRouter>
		);
		expect(container).toMatchSnapshot();
		expect(getByText('Trash Bin')).toBeTruthy();
	});
});
