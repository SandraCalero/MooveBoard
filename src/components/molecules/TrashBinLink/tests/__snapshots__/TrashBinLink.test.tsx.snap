// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TrashBinLink /> should match the snapshot 1`] = `
<div>
  <a
    class="flex flex-col items-center no-underline p-0 m-2 text-center hover:opacity-90 active:translate-y-1"
    href="/trashbin"
  >
    <img
      alt="Trash bin"
      class="w-16 h-16"
      src="trashBinEmpty.png"
    />
    <span>
      Trash Bin
    </span>
  </a>
</div>
`;
